[WIP] New component designs and styleguide: https://sketch.cloud/s/3nq5w

sqevani opened the pull request.

davidyuk commented on build/webpack.conf.js:
```diff
@@ -1,65 +1,176 @@
+    alias: {
+      // Global paths
+      '@source': path.resolve(__dirname, '../src'),
+      '@public': path.resolve(__dirname, '../public'),
```
Why don't use absolute paths (`/src/...` and `/public`) instead of this aliases?

davidyuk commented on package.json:
```diff
@@ -23,9 +23,11 @@
   "dependencies": {
     "ethereum-blockies-png": "git+https://github.com/acdbaykal/ethereum-blockies-png.git",
     "ethjs-unit": "^0.1.6",
+    "normalize-scss": "^7.0.1",
```
Why should we replace `normalize.css` with this package?

davidyuk commented on styleguide.config.js:
```diff
@@ -37,8 +37,8 @@ module.exports = {
+  //require: [
+  //  path.join(__dirname, 'src/components/common.scss')
+  //],
+  assetsDir: 'public'
```
What is the reason for renaming `static` folder?

davidyuk commented on src/components/aeInput/index.js:
```diff
@@ -1,5 +1,8 @@
+ * @param {Vue} Vue
+ * @param {{}} options
+ */
+export const install = function (Vue, options) {
```
`options` is the unused argument, why you have added it?

sqevani commented:
Explanation to what's happening with the big bundle:

- Refactored Webpack and Stylguidist

Webpack now will chunk several things, reduced code generation, minification and other related stuff.

Currentl webpack spits:

1. aepp.components.js, aepp.components.css
- these two are the entire library of components available, and if someone wants to use them simply, imports them and data. Everything is there.

2. aepp.base.css 
- This one is simply the root level CSS that is needed overall for each component to work separately

3. aepp.base.js is just a random file, since which needs to be removed everytime there is a build. (will work on this later.

4. FUTURE: There will be a .js output for each component, and making use of aepp.base.css each component can work independently.

Changed the directories names:

- dist/ -> build/
- build/ -> config/
- created docs/ folder for documentation

Renamed several npm run-scripts names to follow a specific pattern

- `verb`
- `noun` (optional)
- `noun:verb`

```
    "unit": "jest --config config/jest.config.js",
    "unit:watch": "jest --watch",
    "lint": "eslint --ext .js,.vue src",
    "test": "npm run unit && npm run lint",
    "build": "node config/build.js",
    "watch": "node config/watch.js",
    "docs": "vue-styleguidist server --verbose --config config/styleguidist.config.js",
    "docs:build": "vue-styleguidist build --verbose --config config/styleguidist.config.js",
    "prepublish": "npm test",
    "prepare": "npm run build"
```

Also, when it comes to Environmental variables, there's one place to define those. its `.env` and `dotenv` plugin takes care of all the rest.

----

Oh also, size went from: 400KB -> 200KB unminified, minified as for minified it is (drum roll):

<img width="561" alt="screen shot 2018-07-12 at 18 30 36" src="https://user-images.githubusercontent.com/40054938/42625019-ba50daee-8601-11e8-9623-958f3f5f72ed.png">

davidyuk commented on .eslintrc.js:
```diff
@@ -18,11 +18,9 @@ module.exports = {
     // allow debugger during development
-    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0
+    'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
+    "indent": "off"
```
This rule should be enabled.

davidyuk commented on .eslintrc.js:
```diff
@@ -18,11 +18,9 @@ module.exports = {
-      'js': 'never',
-      'vue': 'never'
-    }],
+    'import/extensions': false,
```
What is the reason for this change?

sadiqevani replied:
its like that for now, my IDE had some issues as it didnt recognize the exceptions

davidyuk commented on .githooks/pre-commit/lint:
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+npm run lint
```
I'm against stuff like this: https://github.com/aeternity/aepp-identity/pull/82

sadiqevani replied:
I would like to catch most of the issues before committing and/or pushing.

davidyuk replied:
So, I'm repeating some points from https://github.com/aeternity/aepp-identity/pull/82:
- it is not working properly when developer trying to commit part of the changes of the working directory
- It slows down git performance
- these checks are already performs in CI

davidyuk commented on src/components/ae-toolbar/ae-toolbar.html:
```diff
@@ -0,0 +1,4 @@
+<div class="ae-toolbar" :class="{ [mode]: Boolean(mode), [align]: Boolean(align), minimal, disabled }">
```
`mode` and `align` won't be casted to Boolean automatically?

davidyuk commented on src/components/ae-toolbar/ae-toolbar.js:
```diff
@@ -0,0 +1,58 @@
+          'primary',
+          'secondary'
+        ].indexOf(value) !== -1
+      }
```
```
validator: value => ['primary', 'secondary'].includes(value)
```
?
